extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -64,15 +69,25 @@ class AmpStickyAd extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
-    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
+    // this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return Promise.resolve();
     }
+    // Reschedule layout for ad if layout sticky-ad again.
+    if (this.visible_) {
+      this.updateInViewport(this.ad_, true);
+      this.scheduleLayout(this.ad_);
+    }
     return Promise.resolve();
   }
 
   /** @override */
+  unlayoutCallback() {
```

Comment:
But since we declare ownership for `<amp-ad>` inside, if we don't call `updateInViewport` and `scheduleLayout` for it, The ad itself will never be shown after switching viewer.
Maybe I should define another `needRelayout` variable in `unlayoutCallback()` and only call `scheduleLayout` when `(needRelayout && visible_)` both satisfy